src/main/java/com/github/princesslana/eriscasper/BotContext.java

```diff
@@ -37,13 +38,17 @@ public BotContext(Observable<Event> events, Routes routes, RepositoryManager rep
 
   public <D, E extends Event & Wrapper<D>> Completable on(
       Class<E> evt, Function<D, Completable> f) {
-    return events.ofType(evt).map(e -> e.unwrap()).flatMapCompletable(f::apply);
+    return events.ofType(evt).map(E::unwrap).flatMapCompletable(f::apply);
   }
 
   public Completable doNothing() {
     return Completable.complete();
   }
 
+  public Completable sendGatewayCommand(GatewayCommand command) {
+    return routes.sendGatewayCommand(command);
```

Comment:
I'm not sure this belongs on Routes. Routes is kind of HTTP specific.

But for what reason are we exposing the gateway to the user?

Reply:
When the user wants to do something like send a message, the fact that it's happening over HTTP is hidden from them. There is nothing in the API that refers to rest or http, they do it using the `Action` concept we have.

So, gateway commands can be done in the same way.

Perhaps `Action` is an interface, with implementations `RestAction`, `GatewayAction`.

Reply:
That's kind of impractical if we want the users to know which one to use, we also need to know which one it is because each of them have special attributes. It would cause casting and checking and such if we just wanted to use Action.